src/main/java/org/folio/innreach/domain/event/DomainEvent.java

```diff
@@ -0,0 +1,24 @@
+package org.folio.innreach.domain.event;
+
+import java.util.UUID;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString(exclude = "data")
+public class DomainEvent<T> {
+  protected UUID recordId;
```

Comment:
if it corresponds to "id" field in the event then it's **event id** not record id

Reply:
it corresponds to the identifier of entity (loan) which we set manually from event's key

Reply:
ok